### PR TITLE
Feat/revive

### PR DESCRIPTION
**Pull request by datosh**
### Proposed change(s)

- enable revive as linter

### Additional info
- Originally triggered because I forgot to comment a public function and wanted our linter to notify me. Saw that we have 200 findings and its not a super quick fix. Added to dev sync agenda to decide if we want to fix.

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

File: operators/constellation-node-operator/internal/azure/client/nodeimage_test.go [hunk 1]
```diff
@@ -351,21 +351,21 @@ func TestCapacityPollingHandler(t *testing.T) {
 	assert.Equal(wantCapacity, gotCapacity)
 }
 
-type stubCapacityPoller struct {
```
Comment (katexochen) [inline operators/constellation-node-operator/internal/azure/client/nodeimage_test.go:354]: Imo the stub shouldn't be public, is there anything that returns it?

**Review by katexochen — APPROVED** [review]
LGTM :seal:

File: operators/constellation-node-operator/internal/azure/client/nodeimage_test.go (hunk 1 above)
Comment (daniel-weisse) [reply operators/constellation-node-operator/internal/azure/client/nodeimage_test.go:354]: I think the issues comes from public functions returning private types?
If so `NewStubCapacityPoller` should be made private, if possible, so we don't export testing stubs

Comment (katexochen) [reply operators/constellation-node-operator/internal/azure/client/nodeimage_test.go:354]: That is what I wanted to suggest. :)

**Review by thomasten — APPROVED** [review]

**Pull request merged**